src/main/java/com/redhat/devtools/lsp4ij/LSPIJEditorUtils.java

```diff
@@ -0,0 +1,454 @@
+    //  and would now require reflection to work in 2023.2 and later versions. Specifically it used to be
+    //  "bracePair.left/right" which returned "char", but now it's "bracePair.getLeft()/getRight()" which return
+    //  "CharSequence". I'm leaving the associated changes in but commented out and returning "null" -- existing usages
+    //  will degrade gracefully -- and then when all supported IDE versions have the same interface, this can be
+    //  restored. Perhaps this even prompts removal of support for the oldest versions that have this issue?
+
+    private LSPIJEditorUtils() {
+        // Pure utility class
+    }
+
+    /**
+     * Determines whether or not the provided file is an abstract file type file and is supported by a configured
+     * language server. The latter is important for many editor support behaviors because you don't want to suspend
+     * something like end user typing to wait for a server to start.
+     *
+     * @param file the file
+     * @return true if the file has an abstract file type and is supported by a configured language server;
+     * otherwise false
+     */
+    @ApiStatus.Internal
+    public static boolean isSupportedAbstractFileTypeFile(@NotNull PsiFile file) {
+        return isAbstractFileTypeFile(file) && LanguageServersRegistry.getInstance().isFileSupported(file);
+    }
+
+    /**
+     * Determines whether or not the provided file is a TextMate file and is supported by a configured language server.
+     * The latter is important for many editor support behaviors because you don't want to suspend something like end
+     * user typing to wait for a server to start.
+     *
+     * @param file the file
+     * @return true if the file is TextMate and is supported by a configured language server;
+     * otherwise false
+     */
+    @ApiStatus.Internal
+    public static boolean isSupportedTextMateFile(@NotNull PsiFile file) {
+        return isTextMateFile(file) && LanguageServersRegistry.getInstance().isFileSupported(file);
+    }
+
+    /**
+     * Determines whether or not the provided file is an abstract file type or TextMate file and is supported by a
+     * configured language server. The latter is important for many editor support behaviors because you don't want to
+     * suspend something like end user typing to wait for a server to start.
+     *
+     * @param file the file
+     * @return true if the file is either abstract file type or TextMate and is supported by a configured language
+     * server; otherwise false
+     */
+    @ApiStatus.Internal
+    public static boolean isSupportedAbstractFileTypeOrTextMateFile(@NotNull PsiFile file) {
+        return isSupportedAbstractFileTypeFile(file) || isSupportedTextMateFile(file);
+    }
+
+    private static boolean isAbstractFileTypeFile(@NotNull PsiFile file) {
+        return file.getFileType() instanceof AbstractFileType;
+    }
+
+    private static boolean isTextMateFile(@NotNull PsiFile file) {
```

Comment:
Note that this now checks whether or not the TextMate Bundles plugin is installed and enabled before evaluating `instanceof TextMateFileType`. This avoids a potential `NoClassDefFoundError` is the TextMate Bundles plugin is not installed or enabled.

Reply:
Do you use tbe isTextMatFile that I have merged in the main branch?

Reply:
Just updated it to compare languages only.